community/kernel/src/test/java/org/neo4j/kernel/impl/util/collection/ContinuableArrayCursorTest.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.util.collection;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.function.Supplier;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class ContinuableArrayCursorTest
+{
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void shouldWorkOnEmptySupplier() throws Exception
+    {
+        // given
+        ContinuableArrayCursor cursor = new ContinuableArrayCursor<>( () -> null );
+
+        // then
+        assertFalse( cursor.next() );
+    }
+
+    @Test
+    public void shouldWorkSupplierWithEmptyArray() throws Exception
```

Comment:
Your test names don't say what it means for the `ContiuableArrayCursor` to "work".